matter_server/client/connection.py

```diff
@@ -92,14 +84,20 @@ async def connect(self) -> None:
                 f"a version is required that supports at least api schema {MIN_SCHEMA_VERSION} "
                 " - update the Matter Server to a more recent version."
             )
-        if self.server_info.schema_version > MIN_SCHEMA_VERSION:
-            # server version is higher than expected, log only
+        if info.schema_version > MAX_SCHEMA_VERSION:
+            # server version is too high, raise exception
+            await self._ws_client.close()
+            raise InvalidServerVersion(
+                f"Matter Server schema version is incompatible: {info.schema_version}"
+            )
+        if self.server_info.schema_version != SCHEMA_VERSION:
```

Comment:
Should we even allow this? It sounds like it will just break as data models won't align ?